### PR TITLE
[test] Add initial end to end test

### PR DESCRIPTION
**Pull request by jaylees14**
- Add e2e testing framework - ran via `sbt e2e:test`
- Add e2e test stage on Travis
- Move the generation logic to a separate object - the tests were unable to access the `Main` object for some reason 
- Add a simple e2e test on our `simple.temple` framework. I'll add more in a follow up PR

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -5,8 +5,18 @@ version := "0.1"
 scalaVersion := "2.13.1"
 
 // https://www.scala-sbt.org/1.x/docs/Testing.html#Integration+Tests
+lazy val EndToEndTest = config("e2e") extend(Test)
```
Comment (c-harding) [inline build.sbt:8]: And you tell me not to overload “inheritance”…

**Review by Lewiky — APPROVED** [review]
LGTM

File: build.sbt [hunk 2]
```diff
@@ -5,8 +5,18 @@ version := "0.1"
...
+lazy val endToEndTestSettings: Seq[Def.Setting[_]] =
+  inConfig(EndToEndTest)(Defaults.testSettings) ++
+    Seq(
+      scalaSource in EndToEndTest := baseDirectory.value / "src/e2e/scala",
```
Comment (c-harding) [inline build.sbt:12]: > Adults should have no more than 30g of free sugars a day, (roughly equivalent to 7 sugar cubes).

– NHS.uk (2017). How does sugar in our diet affect our health? [online] Available at: https://www.nhs.uk/live-well/eat-well/how-does-sugar-in-our-diet-affect-our-health/

File: src/e2e/scala/temple/SimpleEndToEndTest.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{FileVisitResult, Files, Path, Paths, SimpleFileVisitor}
+
+import org.scalatest.{FlatSpec, Matchers}
+
+class SimpleEndToEndTest extends FlatSpec with Matchers {
+  behavior of "Temple"
+
+  private def removeDirectory(path: Path): Unit =
+    Files.walkFileTree(
```
Comment (c-harding) [inline src/e2e/scala/temple/SimpleEndToEndTest.scala]: Is there no simple way to recursively delete a directory? Is this your own code, or can we cite it?

**Review by c-harding — APPROVED** [review]
much prettier

**Pull request merged**